backend/src/main/java/ai/verta/modeldb/project/subtypes/CreateProjectHandler.java

```diff
@@ -0,0 +1,192 @@
+    valueMap.put("version_number", newProject.getVersionNumber());
+    valueMap.put("readme_text", newProject.getReadmeText());
+
+    valueMap.put("deleted", false);
+    valueMap.put("created", false);
+    valueMap.put("visibility_migration", true);
+
+    return jdbi.useHandle(
+            handle ->
+                handle.useTransaction(
+                    TransactionIsolationLevel.SERIALIZABLE,
+                    handleForTransaction -> {
+                      String queryString = buildInsertQuery(valueMap, "project");
+
+                      LOGGER.trace("insert project query string: " + queryString);
+                      var query = handleForTransaction.createUpdate(queryString);
+
+                      // Inserting fields arguments based on the keys and value of map
+                      for (Map.Entry<String, Object> objectEntry : valueMap.entrySet()) {
+                        query.bind(objectEntry.getKey(), objectEntry.getValue());
+                      }
+
+                      try {
+                        int count = query.execute();
+                        LOGGER.trace("Project Inserted : " + (count > 0));
+                      } catch (UnableToExecuteStatementException exception) {
+                        // take a brief pause before resubmitting its query/transaction
+                        Thread.sleep(config.getJdbi_retry_time()); // Time in ms
+                        LOGGER.trace("Retry to insert Project");
+                        int count = query.execute();
+                        LOGGER.trace("Project Inserted after retry : " + (count > 0));
+                      }
+
+                      if (!newProject.getTagsList().isEmpty()) {
+                        tagsHandler.addTags(
+                            handleForTransaction, newProject.getId(), newProject.getTagsList());
+                      }
+                      if (!newProject.getAttributesList().isEmpty()) {
+                        attributeHandler.logKeyValues(
+                            handleForTransaction,
+                            newProject.getId(),
+                            newProject.getAttributesList());
+                      }
+                      if (!newProject.getArtifactsList().isEmpty()) {
+                        artifactHandler.logArtifacts(
+                            handleForTransaction,
+                            newProject.getId(),
+                            newProject.getArtifactsList(),
+                            false);
+                      }
+                      if (newProject.getCodeVersionSnapshot().hasCodeArchive()
+                          || newProject.getCodeVersionSnapshot().hasGitSnapshot()) {
+                        codeVersionHandler.logCodeVersion(
+                            handleForTransaction,
+                            newProject.getId(),
+                            false,
+                            newProject.getCodeVersionSnapshot());
```

Comment:
Shouldn't we capture the futures that are being created by the handlers? right now, if we have an error, it's just ignored

Reply:
Here used method are not retuning future because we have passed handler as an argument so if we used future then we lost the handler connection before executing DB operation so above all methods are used single transaction without using future.

Reply:
Oh yeah, sorry.